cmd/generate/config/rules/freemius.go

```diff
@@ -0,0 +1,44 @@
+package rules
+
+import (
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/config/utils"
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/secrets"
+	"github.com/zricethezav/gitleaks/v8/config"
+	"regexp"
+)
+
+func Freemius() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "freemius-secret-key",
+		Description: "Detected a Freemius secret key, potentially exposing sensitive information.",
+		Regex:       regexp.MustCompile(`(?i)["']secret_key["']\s*=>\s*["'](sk_[\S]{29})["']`),
```

Comment:
`[\S]{29}` seems a little too broad, this should probably be `[a-zA-Z0-9]{29}` or `\w{29}` ?

Reply:
but \w{29} wouldn't match this key because it contains characters like `$,@`

`sk_ubb4yN3mzqGR2x8#P7r5&@*xC$utE` ( sample key from their repo https://github.com/Freemius/wordpress-sdk?tab=readme-ov-file#integrating--initializing-the-sdk)

Reply:
i have also updated the tps with this example key

Reply:
I see, I was wondering why this key would currently not be covered by the "generic-api-key" rule. 

Any freemius key that only contains [\w.=-] should already be detected by the generic rule

e.g. ` "secret_key" => "sk_ubb4yN3mzqGR2x8sP7r5dfgxCfutE",` is correctly detected,
but anything containing `# & @ * $` will not be

Unfortunately I could not find any hint in the freemius codebase about whether  `# & @ * $` are the only possible special characters .


Perhaps, since this is a very specific php pattern, to avoid unforeseen false positives, would it make sense to limit this to *.php files?

```golang
r := config.Rule{
...
	Path:        regexp.MustCompile(`(?i)\.php$`),
```

Reply:
> Perhaps, since this is a very specific php pattern, to avoid unforeseen false positives, would it make sense to limit this to *.php files?

You're right, this is only found in php file, i made the changes required.

> side note, the code example in the freemius documentation seems highly irresponsible:

They claim to remove this during deployment (https://freemius.com/help/documentation/wordpress-sdk/integrating-freemius-sdk/)

> Note: The secret key is automatically included in the SDK snippet for convenience. As per the comments in the snippet, this should be removed from production code, but don’t worry if you forget as the PHP processor removes it automatically in the Deployment mechanism

AFAIK they use a PHP parser that relies on a specific AST pattern, so there's always a chance for something to slip through if the snippet structure is changed or if the source code is distributed through another platform.